File: Leetcode-Problems/Find Pivot Index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int prefix = 0;
        int sum = 0;
        for (int i : nums) {
            sum += i;
        }
        for (int i = 0; i < nums.size(); i++) {
            sum -= nums[i];
            if (sum == prefix)
                return i;
            prefix += nums[i];
        }
        return -1;
    }
};

File: Leetcode-Problems/Find Pivot Index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int total = 0;
        for (int i = 0; i < nums.length; i++)
            total += nums[i];
        int running_sum = 0;
        for (int i = 0; i < nums.length; i++) {
            if (running_sum == total - running_sum - nums[i]) {
                return i;
            }
            running_sum += nums[i];
        }
        return -1;
    }
}